packages/react/src/components/groups/example.tsx

```diff
@@ -0,0 +1,47 @@
+import { Groups } from './index'
+
+/**
+ * In this example utility functions are attached
+ * to the component constructor as a static method
+ * 
+ * We do this because without submodules it's otherwise
+ * difficult to export related code without congesting
+ * the root namespace.
+ */
+const Util = () => {
+  return (
+    <Groups
+      belongsTo={"hello"}
+
+      belongsTo={Groups.any(["hello", "world"])}
+
+      belongsTo={Groups.all(["hello", "world"])}
+
+      belongsTo={groups => ["some", "custom", "algorithm"]}
+    >
+      hello world
+    </Groups>
+  )
+}
+
+/**
+ * In this example there is a disciminated union in the
+ * component props which notify typescript users when
+ * they break the rules.
+ * 
+ * For javascript users there is a runtime error:
+ * "Only one allowed: belongsTo, belongsToAny, belongsToAll"
+ */
+const Union = () => {
+  return (
+    <Groups
+      belongsTo={"hello"}
+
+      belongsToAny={["hello", "world"]}
+
+      belongsToAll={["hello", "world"]}
+    >
+      hello world
+    </Groups>
+  )
+}
```

Comment:
Please see these prototypes!

Reply:
I can't rely on anything but personal preference here - I much prefer the Util example:
- smaller API surface
- no need to assert only a single mutually exclusive prop was used
- no implicit behaviour
- no room for typos (`belongsToAny` and `belongsToAll` might be swapped mistakenly)

I also don't know what the most frequent use cases are going to be, but I do have a feeling that the most commonly used option would be just specifying a single group name.

Reply:
I would prefer to have them exported like
```
import { Groups } from '@slashid/react'
import { any, all } from '@slashid/react/groups'

// or

import { Groups, any, all } from '@slashid/react'
```

but this feels like a pretty good compromise